engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java

```diff
@@ -511,6 +512,17 @@ private Duration getTemporalValidationTolerance(ConfigurationState configuration
 		return Duration.ZERO;
 	}
 
+	private Object getConstraintValidatorPayload(ConfigurationState configurationState, Map<String, String> properties) {
+		if ( configurationState instanceof ConfigurationImpl ) {
+			ConfigurationImpl hibernateSpecificConfig = (ConfigurationImpl) configurationState;
+			if ( hibernateSpecificConfig.getConstraintValidatorPayload() != null ) {
+				LOG.logConstraintValidatorPayload( hibernateSpecificConfig.getConstraintValidatorPayload() );
```

Comment:
I'm reducing the log level of this one to `DEBUG` as it might be a bit annoying to have the payload object logged (typically in your use case, if the context object has a valid `toString()` that is a bit long, it's going to be a pain). As we don't have the properties in this case, the value being used is pretty clear: it's the one set.